python/doc/examples/getting_started/plot_kriging_vs_gpr.py

```diff
@@ -0,0 +1,507 @@
+
+# %%
+kriging_algo = ot.KrigingAlgorithm(x_train, y_train, covarianceModel, basis)
+kriging_algo.run()
+kriging_result = kriging_algo.getResult()
+krigeageMM = kriging_result.getMetaModel()
+
+# %%
+# We observe that the `scale` and `amplitude` hyper-parameters have been optimized by the `run` method.
+# The used optimization method (default one) is the :class:`~openturns.TNC`
+# With the new API, the :class:`~openturns.experimental.GaussianProcessFitter` class  is used to fit the
+# gaussian process and :class:`~openturns.experimental.GaussianProcessRegression` to get the conditioned model.
+
+# %%
+fitter_algo = otexp.GaussianProcessFitter(x_train, y_train, covarianceModel, basis)
+fitter_algo.run()
+fitter_result = fitter_algo.getResult()
+gpr_algo = otexp.GaussianProcessRegression(fitter_result)
+gpr_algo.run()
+gpr_result = gpr_algo.getResult()
+gprMetamodel = gpr_result.getMetaModel()
+
+# %%
+# We observe that the `scale` and `amplitude` hyper-parameters have been optimized by the :meth:`~openturns.experimental.GaussianProcessFitter.run` method.
+# The used optimization method (default one for this class) is the :class:`~openturns.Cobyla`, which is different from the old API.
+# Then we get the metamodel with `getMetaModel` for evaluating the outputs of the metamodel on the test design of experiments.
+
+# %%
+# Now we plot Gaussian process Regression output, in addition to the previous plots
+
+# %%
+graph = ot.Graph("Comparison data vs GP models", "", "", True, "")
+graph.add(plot_1d_data(x_test, y_test, legend="Exact", color="black", linestyle="dashed"))
+graph.add(plot_1d_data(x_train, y_train, type="Cloud", legend="Data", color="red"))
+graph.add(plot_1d_data(x_test, krigeageMM(x_test), legend="Kriging", color="blue"))
+graph.add(plot_1d_data(x_test, gprMetamodel(x_test), legend="GPR", color="green"))
+graph.setAxes(True)
+graph.setXTitle("X")
+graph.setYTitle("Y")
+graph.setLegendPosition("upper right")
+view = viewer.View(graph)
+
+# %%
+# We see that the Gaussian process regression estimated with both classes is interpolating.
+# This is what is meant by *conditioning* a Gaussian process.
+# We see that, when the sine function has a strong curvature between two points which are separated by a large distance (e.g. between :math:`x=4` and :math:`x=6`),
+# then the gaussian regression is not close to the function :math:`g`.
+# However, when the training points are close (e.g. between :math:`x=11` and :math:`x=11.5`) or when the function is nearly linear
+# (e.g. between :math:`x=8` and :math:`x=11`),
+# then the gaussian process regression is quite accurate.
+
+# %%
+# Activating nugget factor
+# ------------------------
+# Both APIs allow one to estimate the model with an active nugget factor thanks to the :meth:`~openturns.CovarianceModel.activateNuggetFactor`,
+# e.g. the parameter is estimated within the optimization process.
+#
```

Comment:
IMO, a paragraph is missing here to explain that in this part, we introduce a noise to data point. This noise is generated through i.i.d. realizations of a Normaly distributed aleatory variable that is summed to the real function.